File: ios/Runner/TableView/Model/Movie.swift
//
//  Movie.swift
//  Runner
//
//  Created by mateus.santos on 27/04/21.
//

import Foundation

protocol Movie {
    var id:Int { get }
    var title:String { get }
    var overview:String { get }
    var poster:String { get }
    var releaseDate:String { get }
}

struct MovieElement:Movie {
    var id:Int
    var title:String
    var overview:String
    var poster:String
    var releaseDate:String
    
    init() {
        self.id = 0
        self.title = "ERROR"
        self.overview = "ERROR"
        self.poster = "ERROR"
        self.releaseDate = "ERROR"
    }
}

File: add-to-app/data_plugin/lib/Movie.dart
import 'dart:convert';

import 'dart:ffi';

class Movie {
  final int id;
  final String title;
  final String path;
  final String overview;
  final String release_date;

  Movie(this.id, this.title, this.path, this.overview, this.release_date);
}
